src/decoder/mod.rs

```diff
@@ -238,6 +283,12 @@ fn rev_hpredict(image: DecodingBuffer, size: (u32, u32), color_type: ColorType)
         DecodingBuffer::U64(buf) => {
             rev_hpredict_nsamp(buf, size, samples)?;
         }
+        DecodingBuffer::F32(buf) => {
+            rev_hpredict_nsamp(buf, size, samples)?;
+        }
+        DecodingBuffer::F64(buf) => {
+            rev_hpredict_nsamp(buf, size, samples)?;
+        }
```

Comment:
Better not to support those until we have right test data, I think. Having the data types and sample format checks is already a good part of the change.

Reply:
So you think we should just remove this and release floating point support ? And make another PR latter to add the support for the decoding buffer ?

Reply:
Yes, not supporting the prediction seems less painful than a likely buggy implementation.